pysph/tools/ipython.py

```diff
@@ -0,0 +1,523 @@
+
+        self.scalar = widgets.Dropdown(
+            options=[
+                'None'
+            ]+particlearray.output_property_arrays,
+            value='rho',
+            description="scalar",
+            disabled=False,
+            layout=widgets.Layout(width='240px', display='flex')
+        )
+        self.scalar.owner = particles_type
+        self.legend = widgets.ToggleButton(
+            value=False,
+            description="legend",
+            disabled=False,
+            tooltip='Description',
+            layout=widgets.Layout(width='80px', display='flex')
+        )
+        self.legend.owner = particles_type
+        self.vector = widgets.Text(
+            value='',
+            placeholder='variable1,variable2',
+            description='vector',
+            disabled=False,
+            layout=widgets.Layout(width='240px', display='flex')
+        )
+        self.vector.owner = particles_type
+        self.vector_width = widgets.FloatSlider(
+            min=1,
+            max=100,
+            step=1,
+            value=25,
+            description='vector width',
+            layout=widgets.Layout(width='300px'),
+        )
+        self.vector_width.owner = particles_type
+        self.vector_scale = widgets.FloatSlider(
+            min=1,
+            max=100,
+            step=1,
+            value=55,
+            description='vector scale',
+            layout=widgets.Layout(width='300px'),
+        )
+        self.vector_scale.owner = particles_type
+        self.scalar_size = widgets.FloatSlider(
+            min=0,
+            max=50,
+            step=1,
+            value=10,
+            description='scalar size',
+            layout=widgets.Layout(width='300px'),
+        )
+        self.scalar_size.owner = particles_type
+
+class Viewer2DWidgets(object):
+
+    def __init__(self, file, file_count):
+
+        self.temp_data = load(file)['arrays']
+        self.frame = widgets.IntSlider(
+            min=0,
+            max=file_count,
+            step=1,
+            value=0,
+            description='frame',
+            layout=widgets.Layout(width='600px'),
+        )
+        self.save_figure = widgets.Text(
+                value='',
+                placeholder='example.pdf',
+                description='Save figure',
+                disabled=False,
+                layout=widgets.Layout(width='240px', display='flex')
+        )
+        self.particles = {}
+        for particles_type in self.temp_data.keys():
+            self.particles[particles_type] = ParticleArrayWidgets(
+                self.temp_data[particles_type],
+                particles_type,
+            )
+
+    def _create_vbox(self):
+
+        from ipywidgets import HBox, VBox, Label, Layout
+        items = []
+        for particles_type in self.particles.keys():
+            pa_widgets = self.particles[particles_type]
+            items.append(
+                VBox([
```

Comment:
Can the vbox also be pushed to the `ParticleArrayWidgets`?  It smells that you are accessing the attributes of it to create a vbox here.